src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesHttpAsyncAwsTransport.php

```diff
@@ -65,6 +65,11 @@ protected function doSend(SentMessage $message): void
 
             throw $exception;
         }
+
+        if ($message->getOriginalMessage() instanceof Message) {
+            $message->getOriginalMessage()->getHeaders()->addHeader('X-Message-ID', $result->getMessageId());
+            $message->getOriginalMessage()->getHeaders()->addHeader('X-SES-Message-ID', $result->getMessageId());
```

Comment:
It feels wrong to modify the original message. This should be part of the SendMessage.
(I've just read the related issue and I agree that this fix is probably not the right one).

Reply:
@kbond I would love to get your thoughts.

Reply:
Heh, same conclusion @driesvints came to in #45871. The other, more complex idea was https://github.com/symfony/symfony/issues/45871#issuecomment-1080683423.